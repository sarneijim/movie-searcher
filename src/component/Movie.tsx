import React, { Component } from "react";
import Typography from "@material-ui/core/Typography";
import { Link } from "react-router-dom";
import styles from "../styles/details.module.css";
import { imgPath, movieRequest } from "../request";

export default class Movie extends Component<{}, any> {
  constructor(props: any) {
    super(props);
    this.state = { data: {} };
    if (props.match?.params?.id) {
      this.httpRequest(props.match.params.id);
    }
  }

  httpRequest = async (id: string) => {
    try {
      var response = await movieRequest(id, {
        append_to_response: "credits",
      });
      this.setState({ data: response });
    } catch (err) {
      console.error(err);
    }
  };

  render() {
    return (
      <div className={styles.root}>
        <div className={styles.movie}>
          <div className={styles.left}>
            <img
              className={styles.img}
              src={`${imgPath}${this.state.data.poster_path}`}
              alt={this.state.data.title}
            />
            <span
              className={`${styles.marker}
                ${
                  this.state.data.vote_average >= 7 ? styles["marker-good"] : ""
                } 
                ${
                  this.state.data.vote_average < 5 ? styles["marker-bad"] : ""
                }`}
            >
              {this.state.data.vote_average}
            </span>
          </div>
          <div className={styles.right}>
            <Typography component="h4" variant="h4">
              {this.state.data.title}
            </Typography>
            <Typography variant="subtitle1" color="textSecondary">
              {this.state.data.overview}
            </Typography>
          </div>
        </div>
        <div className={styles.cast}>
          <Typography component="h5" variant="h5">
            Cast
          </Typography>
          <div className={styles.gallery}>
            {this.state.data.credits?.cast.map((actor: any) => (
              <div key={actor.id} className={styles.actor}>
                <Link to={`/person/${actor.id}`}>
                  <img
                    className={styles.img}
                    src={`${imgPath}${actor.profile_path}`}
                    alt={actor.title}
                  />
                  <div className={styles["actor-name"]}>
                    <span>{actor.name}</span>
                  </div>
                  <div className={styles["character-name"]}>
                    <span>{actor.character}</span>
                  </div>
                </Link>
              </div>
            ))}
          </div>
        </div>
      </div>
    );
  }
}
