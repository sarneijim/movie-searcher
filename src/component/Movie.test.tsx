import renderer from "react-test-renderer";
import Movie from "./Movie";

describe("Movie", () => {
  let component: any;
  beforeEach(() => {
    component = renderer.create(<Movie></Movie>);
  });

  afterEach(() => {
    component = null;
  });

  it("should create", () => {
    expect(component).toBeTruthy();
  });

  it("should match the snapshot", () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("check httpRequest method", () => {
    let instance: any = component.getInstance();
    const utils = require("../request");
    const movieRequestSpy = jest
      .spyOn(utils, "movieRequest")
      .mockReturnValue({ id: 123 });
    instance.httpRequest("abc");
    expect(movieRequestSpy).toHaveBeenCalledWith("abc", {
      append_to_response: "credits",
    });
    movieRequestSpy.mockClear();
  });
});
