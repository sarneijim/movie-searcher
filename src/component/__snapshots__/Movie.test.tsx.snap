// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should match the snapshot 1`] = `
<div
  className="root"
>
  <div
    className="movie"
  >
    <div
      className="left"
    >
      <img
        className="img"
        src="https://image.tmdb.org/t/p/w500undefined"
      />
      <span
        className="marker
                 
                "
      />
    </div>
    <div
      className="right"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4"
      />
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
      />
    </div>
  </div>
  <div
    className="cast"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Cast
    </h5>
    <div
      className="gallery"
    />
  </div>
</div>
`;
